plugins/modules/idrac_network_attributes.py

```diff
@@ -0,0 +1,534 @@
+                self.network_adapter_id_uri = each_adapter.get(odata, '')
+                break
+        if not found_adapter:
+            self.module.exit_json(failed=True, msg=INVALID_ID_MSG.format(network_adapter_id,
+                                                                         'network_adapter_id'))
+
+    def perform_validation_for_network_device_function_id(self):
+        odata = '@odata.id'
+        found_device = False
+        network_device_function_id = self.module.params.get('network_device_function_id')
+        network_devices = get_dynamic_uri(self.idrac, self.network_adapter_id_uri, 'NetworkDeviceFunctions')[odata]
+        network_device_list = get_dynamic_uri(self.idrac, network_devices, 'Members')
+        for each_device in network_device_list:
+            if network_device_function_id in each_device.get(odata, ''):
+                found_device = True
+                self.network_device_function_id_uri = each_device.get(odata, '')
+                break
+        if not found_device:
+            self.module.exit_json(failed=True, msg=INVALID_ID_MSG.format(network_device_function_id,
+                                                                         'network_device_function_id'))
+
+    def validate_job_timeout(self):
+        if self.module.params.get("job_wait") and self.module.params.get("job_wait_timeout") <= 0:
+            self.module.exit_json(msg=TIMEOUT_NEGATIVE_OR_ZERO_MSG, failed=True)
+
+    def apply_time(self, setting_uri):
+        resp = get_dynamic_uri(self.idrac, setting_uri, "@Redfish.Settings")
+        rf_settings = resp.get("SupportedApplyTimes", [])
+        apply_time = self.module.params.get('apply_time', {})
+        rf_set, reboot_required = self.__get_redfish_apply_time(apply_time, rf_settings)
+        return rf_set
+
+
+class OEMNetworkAttributes(IDRACNetworkAttributes):
+    def __init__(self, idrac, module, base_uri):
+        super().__init__(idrac, module, base_uri)
+
+    def clear_pending(self):
+        oem_links = get_dynamic_uri(self.idrac, self.network_device_function_id_uri, 'Links')
+        oem_uri = oem_links.get('Oem').get('Dell').get('DellNetworkAttributes').get('@odata.id')
+        resp = get_dynamic_uri(self.idrac, oem_uri, '@Redfish.Settings')
+        settings_uri = resp.get('SettingsObject').get('@odata.id')
+        settings_uri_resp = get_dynamic_uri(self.idrac, settings_uri)
+        pending_attributes = settings_uri_resp.get('Attributes')
+        clear_pending_uri = settings_uri_resp.get('Actions').get('#DellManager.ClearPending').get('target')
+        if not pending_attributes:
+            self.module.exit_json(msg=NO_CHANGES_FOUND_MSG)
+        job_resp = get_scheduled_job_resp(self.idrac, 'NICConfiguration')
+        job_id, job_state = job_resp.get('Id'), job_resp.get('JobState')
+        if job_id:
+            if job_state in ["Running"]:
+                job_resp = remove_key(job_resp, regex_pattern='(.*?)@odata')
```

Comment:
remove_key already has @odata by default.

Reply:
This pattern will remove anywhere odata mentioned in key, start, between, last. I think this is more appropriate.., better to keep this.